client/pages/sections/protocols/protocols.js

```diff
@@ -0,0 +1,128 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import classnames from 'classnames';
+
+import size from 'lodash/size';
+import ProtocolSections from './protocol-sections';
+
+import Fieldset from '../../../components/fieldset';
+import Repeater from '../../../components/repeater';
+import Controls from '../../../components/controls';
+
+class Form extends PureComponent {
+  removeItem = e => {
+    e.preventDefault();
+    this.props.removeItem();
+  }
+
+  render() {
+    const { index, name, updateItem, exit, toggleActive, prefix = '', ...props } = this.props;
+    return (
+      <div className={classnames('protocol', 'panel')}>
+        {
+          index !== 0 && (
+            <a href="#" className="float-right" onClick={this.removeItem}>Remove</a>
+          )
+        }
+
+        <h2>{`Protocol ${index + 1}`}</h2>
+        <Fieldset
+          { ...props }
+          fields={props.fields}
+          prefix={`${prefix}${name}-${index}-`}
+          onFieldChange={(key, value) => updateItem({ [key]: value })}
+        />
+        <Controls onContinue={toggleActive} onExit={exit} exitDisabled={true} />
+      </div>
+    );
+  }
+}
+
+class Protocol extends PureComponent {
+  state = {
+    active: !this.props.values.title,
+    complete: false
+  }
+
+  toggleActive = () => {
+    this.setState({ active: !this.state.active });
+  }
+
+  getParts = () => {
+    if (!this.props.editable || !this.props.location.hash) {
+      return null;
+    }
+    const parts = this.props.location.hash.split('.');
+    if (!parts.length) {
+      return null;
+    }
+    return parts;
+  }
+
+  isActive = parts => {
```

Comment:
The only time we call this function we have called `this.getParts` immediately before to find the argument. 

I'd probably move that call to inside this function and do away with the argument here for marginally improved efficiency.

Reply:
See edit.